docs/openshift-on-azure-walkthrough.md

```diff
@@ -195,11 +201,127 @@ oc login <FQDN> --token=<YOUR_TOKEN>
 ```
 
 
+## AAD application configuration
+
+The deployed OpenShift cluster needs a valid AAD application and service principal to call back into the Azure API,
+in order to enable AAD authentication. There are a few options here:
+
+* Create application using Azure Portal and configure all values there.
+
+* Create application using CLI and later configure permission in the Azure portal.
+
+* Create application using CLI and approve permission in the Azure portal
+
+### Option 1: Create application using Azure portal
+
+Do a search for `App registrations` in the search section located at the top and navigate to it or go to AAD -> app registrations
+
+![](./medias/OSA_APP_Portal.png)
+
+Press `New application registration` button. Enter details for your application instance and press `Create`.
+
+![](./medias/OSA_AAD_Portal_Create.png)
+
+ Sign-on URL is OpenShift web console URL: `https://awesome-cluster.eastus.cloudapp.azure.com/oauth2callback/Azure%20AD`
+
+Copy value of the `Application ID` fields for `OSA_AAD_ID` from [step 4 above](#Step-4:-Create-OpenShift-cluster). 
+
+![](./medias/OSA_AAD_Portal_Details.png)
+
+`OSA_AAD_SECRET` Value should be set to `Key`, attached to the AAD application. Under same AAD application press `Keys` and create new key:
+
+![](./medias/OSA_AAD_Portal_key_create.png)
+
+> Once this key is created and the page is left the value will be hidden. Make sure to store is somewhere safe.
+
+Key value is used for `az openshift create` command `--aad-client-app-secret <aad key value>`
+
+Next you will need to grant set of API permissions for sign-on to work. Under same application press `Required permissions` and `Add`. Set these values:
+
+```
+Microsoft Graph:
+  Application permissions: "Read all groups", "Read directory data"
+  Delegated permissions: "Read all groups", "Sign users in"
+Windows Azure Active Directory:
+  Delegated permissions: "Sign in and read user profile"
+```
+
+For the permissions to take effect, Azure subscription administrator will need to press `Grant permissions` to approve permission request.
+
+### Option 2: Create application using CLI and later configure permission in the Azure portal
+
+To create application using azure CLI execute:
+
+
+```azurecli-interactive
+OSA_AAD_SECRET=MyAw3s0meP@ssw0rd!
+OSA_AAD_REPLY_URL=https://$OSA_CLUSTER_NAME.$LOCATION.cloudapp.azure.com/oauth2callback/Azure%20AD
+
+az ad app create --display-name $OSA_CLUSTER_NAME --key-type Password --password $OSA_AAD_SECRET --identifier-uris $OSA_AAD_REPLY_URL --reply-urls $OSA_AAD_REPLY_URL
+```
+
+You will be prompted with the output of the newly created application. After this you will need to configure API permissions for the created application. Follow option 1 from the place, where it speaks about granting set of API permissions for sign-on to work.
```

Comment:
Also I think this needs to be in bold letters or something, make it clear that *just* creating the app via cli tool is still not enough, I feel it's going to be easy to miss this